app/serializers/alchemy/page_tree_serializer.rb

```diff
@@ -55,6 +55,7 @@ def page_hash(page, has_children, level, folded)
         public: page.public?,
         restricted: page.restricted?,
         page_layout: page.page_layout,
+        child_pages_allowed: page.definition["child_pages"] != false,
```

Comment:
I do not like the name of this attribute 

`allowed_page_types`, or maybe `nestable_pages` `nestable_child_pages`?

Reply:
Now i named it for what it actually is `show_create_page_btn`. Which can be true even if there are no pages available, as the button is only hidden if `definition['child_pages'] == false`.

Also added tests for controller method calls. I've been using this feature for quite some time already, `[space] &.` works just as fine as `[no_space].&` - but i still removed the space as you like.

I think the implementation is good now and for me the feature is essential for controlling the page tree structure. It is also just as you requested it to be as explained in this comment https://github.com/AlchemyCMS/alchemy_cms/pull/1776#issuecomment-651109428

I would say it cannot be simplified more. You have to give me more concrete feedback if you still disagree.